### PR TITLE
Adds a CDash config file

### PR DESCRIPTION
**Pull request by kennyweiss**
# Summary

- This PR is a feature
- It adds a cdash config file for our LC configurations
- It also adds a host-config that does not require any third-party libraries

### TODO
- ~Incorporate into our gitlab-ci.yaml file ?~  
  We'll incorporate this into `gitlab-ci.yaml` if a follow-up PR.

### CODE REVIEW

**Review by rhornung67 — APPROVED** [review]

**Review by agcapps — APPROVED** [review]
All right, let's give this a try.  Thank you, @kennyweiss .

File: src/cmake/CMakeBasics.cmake [hunk 1]
```diff
@@ -206,3 +206,42 @@ endif()
...
+    # e.g. in our MacOS CI AppleClang configuration
+    if(NOT CMAKE_CONFIGURATION_TYPES AND NOT "${CMAKE_BUILD_TYPE}" STREQUAL "")
+        blt_list_append(TO _config ELEMENTS ${CMAKE_BUILD_TYPE})
+    endif()
+    blt_list_append(TO _config ELEMENTS "shared" IF BUILD_SHARED_LIBS)
+    blt_list_append(TO _config ELEMENTS "cuda" IF ENABLE_CUDA)
+    blt_list_append(TO _config ELEMENTS "hip" IF ENABLE_HIP)
+    blt_list_append(TO _config ELEMENTS "mpi" IF ENABLE_MPI)
+    blt_list_append(TO _config ELEMENTS "openmp" IF ENABLE_OPENMP)
+    string(REPLACE ";" "-" AXOM_CONFIG_NAME "${_config}")
```
Comment (white238) [inline src/cmake/CMakeBasics.cmake:238]:
```suggestion
    list(JOIN _config "-" AXOM_CONFIG_NAME)
```

More readable imo.

File: src/cmake/Dashboard.cmake.in [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+set(CTEST_CUSTOM_MAXIMUM_NUMBER_OF_ERRORS           "200" )
+set(CTEST_CUSTOM_MAXIMUM_NUMBER_OF_WARNINGS         "500" )
+set(CTEST_CUSTOM_MAXIMUM_PASSED_TEST_OUTPUT_SIZE    "104857600") # 100 MB
+set(CTEST_CUSTOM_COVERAGE_EXCLUDE                   "")
+set(CTEST_NIGHTLY_START_TIME                        "01:00:00 UTC")
+set(CTEST_USE_LAUNCHERS                             1)
+
+# list of additional warning messages to filter
+list(APPEND CTEST_CUSTOM_WARNING_EXCEPTION
```
Comment (white238) [inline src/cmake/Dashboard.cmake.in:30]: thousand thumbs up. thanks for figuring this out.

**Review by white238 — APPROVED** [review]

File: src/cmake/CMakeBasics.cmake (hunk 1 above)
Comment (kennyweiss) [reply src/cmake/CMakeBasics.cmake:238]: Thanks -- I wasn't aware of `list(JOIN ...)`

File: src/cmake/Dashboard.cmake.in (hunk 2 above)
Comment (rhornung67) [reply src/cmake/Dashboard.cmake.in:30]: Wow. 10 x 💯 👍

**Pull request merged**